File: Python/_0189_Rotate_Array/Solution.py
# coding=utf-8
# @Time       : 2020/3/8
# @Author     : Wang Xiaoxiao
# @University : Dalian University of Technology
# @FileName   : Solution.py
# @Software   : PyCharm
# @github     : https://github.com/i-love-linux/LeetCode
# 旋转数组


class Solution(object):
    def rotate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        if k % len(nums) == 0:
            return

        k = k % len(nums)
        nums.reverse()
        nums[:k] = list(reversed(nums[:k]))
        nums[k:] = list(reversed(nums[k:]))


if __name__ == "__main__":
    nums = [1, 2, 3, 4, 5, 6, 7]
    k = 3
    So = Solution()
    So.rotate(nums, k)
    print(nums)

File: Java/src/_0189_Rotate_Array/Solution.java
package _0189_Rotate_Array;

import _0000_study._common.Output;

/**
 * 暴力解法
 * 时间复杂度：O(n*k)
 * 空间复杂度：O(1)
 */
public class Solution {

    public void rotate(int[] nums, int k) {

        if (k % nums.length == 0) return;

        for (int i = 0; i < k; i++) {
            int temp = nums[nums.length - 1];
            for (int j = nums.length - 1; j > 0; j--)
                nums[j] = nums[j - 1];
            nums[0] = temp;
        }
    }

    public static void main(String[] args) {

        int[] nums = {1, 2, 3, 4, 5, 6, 7};
        int k = 3;
        (new Solution()).rotate(nums, k);
        Output.OutputBasicArray1D(nums);
    }
}
